### PR TITLE
io/lab: Explain the need for loops when using read/write

### PR DESCRIPTION
**Pull request by Alex-deVis**
Explain the need for loops when using `read()` and `write()` and remove manual quotations.

### CODE REVIEW

File: content/chapters/io/lab/content/file-descriptors.md [hunk 1]
```diff
@@ -70,14 +70,13 @@ It returns the number of bytes that were read from the file.
...
-The reason is, as always, stated in [`read()`'s man page](https://man7.org/linux/man-pages/man2/read.2.html).
-The snippet below is about the number of bytes returned by `read()`:
 
-> It is not an error if this number is smaller than the number of
-> bytes requested; this may happen for example because fewer bytes
-> are actually available right now (maybe because we were close to
-> end-of-file, or because we are reading from a pipe, or from a
-> terminal), or because read() was interrupted by a signal.
+The return value of `read()` may differ from _bytes_to_read_ if there are not enough bytes available or if the operation is interrupted by a signal.
+A return value between **0** and _bytes_to_read_ is not enough to decide whether we should stop reading.
```
Comment (teodutu) [inline content/chapters/io/lab/content/file-descriptors.md:75]:
```suggestion
The return value of `read()` may differ from `bytes_to_read` if there are not enough bytes available or if the operation is interrupted by a signal.
A return value between **0** and `bytes_to_read` is not enough to decide whether we should stop reading.
```

File: content/chapters/io/lab/content/file-descriptors.md [hunk 2]
```diff
@@ -224,11 +223,11 @@ For a quick recap of the flags we've discussed so far, take a look at the follow
...
-| `"w+"`         | `O_RDWR │ O_CREAT │ O_TRUNC`        |
-| `"a+"`         | `O_RDWR │ O_CREAT │ O_APPEND`       |
+| `fopen()` mode |          `open()` flag          |
+| :------------: | :-----------------------------: |
+|     `"r"`      |           `O_RDONLY`            |
+|     `"w"`      | `O_WRONLY │ O_CREAT │ O_TRUNC`  |
+|     `"a"`      | `O_WRONLY │ O_CREAT │ O_APPEND` |
+|     `"r+"`     |            `O_RDWR`             |
+|     `"w+"`     |  `O_RDWR │ O_CREAT │ O_TRUNC`   |
+|     `"a+"`     |  `O_RDWR │ O_CREAT │ O_APPEND`  |
```
Comment (teodutu) [inline content/chapters/io/lab/content/file-descriptors.md:233]: attaboy

File: content/chapters/io/lab/content/file-descriptors.md [hunk 3]
```diff
@@ -70,14 +70,13 @@ It returns the number of bytes that were read from the file.
...
 
-> It is not an error if this number is smaller than the number of
-> bytes requested; this may happen for example because fewer bytes
-> are actually available right now (maybe because we were close to
-> end-of-file, or because we are reading from a pipe, or from a
-> terminal), or because read() was interrupted by a signal.
+The return value of `read()` may differ from _bytes_to_read_ if there are not enough bytes available or if the operation is interrupted by a signal.
+A return value between **0** and _bytes_to_read_ is not enough to decide whether we should stop reading.
+To determine this, we make another `read()` call, which will return **0** if the cursor is already at **EOF** (end of file).
+
```
Comment (teodutu) [inline content/chapters/io/lab/content/file-descriptors.md:77]: I would also add a link to [the "RETURN VALUE" section from `read()`'s man page](https://man7.org/linux/man-pages/man2/read.2.html#RETURN_VALUE)

**Review by teodutu — CHANGES_REQUESTED** [review]
Just a few nits and then it'll be alright.

File: content/chapters/io/lab/content/file-descriptors.md [hunk 4]
```diff
@@ -70,14 +70,13 @@ It returns the number of bytes that were read from the file.
...
 Instead, it uses a `while` loop to read data from the file.
-The reason is, as always, stated in [`read()`'s man page](https://man7.org/linux/man-pages/man2/read.2.html).
-The snippet below is about the number of bytes returned by `read()`:
 
-> It is not an error if this number is smaller than the number of
-> bytes requested; this may happen for example because fewer bytes
-> are actually available right now (maybe because we were close to
-> end-of-file, or because we are reading from a pipe, or from a
-> terminal), or because read() was interrupted by a signal.
+The [return value of `read()`](https://man7.org/linux/man-pages/man2/read.2.html#RETURN_VALUE) may differ from `bytes_to_read` if there are not enough bytes available or if the operation is interrupted by a signal.
```
Comment (teodutu) [inline content/chapters/io/lab/content/file-descriptors.md:74]:
```suggestion
The [return value of `read()`](https://man7.org/linux/man-pages/man2/read.2.html#RETURN_VALUE) may be less than `bytes_to_read` if there are not enough bytes available or if the operation is interrupted by a signal.
```

**Review by teodutu — CHANGES_REQUESTED** [review]
One more take and it's done.

**Comment by Alex-deVis** [comment]
@teodutu, I applied your suggestion, this is ready to be squashed and merged.

**Review by teodutu — APPROVED** [review]

**Pull request merged**